### PR TITLE
Add and Remove function are add to project

### PR DESCRIPTION
**Pull request by Nizamuddin4493**
# Add and Remove functions are added to the project. :rocket: 
- Create a new JavaScript file for the new functionality.
- Implemented a function for adding a new task (add a new element to the array).
- Implemented a function for editing task descriptions.
- Implemented a function for deleting a task (remove an element from the array).
- Implemented a function for the "Clear all completed" button (use filter() method).

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -7,9 +7,13 @@ import search from './images/searching.png';
...
+import trash from './images/trash.png';
 import {
   itemsArr, tasksListShow, getItems, saveLocalStorage,
 } from './activeList.js';
+import {
+  reload, removeAll, removeItem, updateItem,
+} from './AddRemove.js';
 
 const items = [
   {
```
Comment (EricMbouwe) [inline src/index.js]: - [ ] All hardcoded items from the tasks array should be removed; Let's fix that

File: src/activeList.js [hunk 2]
```diff
@@ -36,8 +89,9 @@ const tasksListShow = (itemsArr) => {
       parentNode.insertAdjacentHTML('beforeend',
         `<div class="show-list">
         <input type="checkbox" data-done class="check-box">
-      <p class="tast" data-time="${item.index}">${item.description}</p>
+      <p class="tast" data-edit data-time="${item.index}">${item.description}</p>
       <img src="images/drag-drop.png" alt="" class="drag-drop">
+      <img src="images/trash.png" alt="" data-delete class="trash">
       </div>`);
       handleItem(item);
     });
```
Comment (EricMbouwe) [inline src/activeList.js]: - [ ] After selecting some task your app keep the good state in the localStorage after a page reload but the UI state is not kept, the selected tasks are become unselected. Let's fix that issue

File: src/activeList.js [hunk 3]
```diff
@@ -22,9 +43,41 @@ const handleItem = function (itemData) {
...
+          item.querySelector('.tast').style.textDecoration = 'line-through';
+          item.querySelector('.drag-drop').style.display = 'none';
+          item.querySelector('.trash').style.display = 'flex';
+        }
+        removeAll.addEventListener('click', (e) => {
+          e.preventDefault();
+          // eslint-disable-next-line eqeqeq
+          if (currentItem.completed == true) {
+            const removeIndex = itemsArr.indexOf(currentItem);
+            itemsArr.splice(removeIndex, 1);
```
Comment (EricMbouwe) [inline src/activeList.js:62]: - [ ] You should have A **function** for the "Clear all completed" button using the `Array.filter()` method. Let's fix that

**Review by EricMbouwe — CHANGES_REQUESTED** [review]
### Hi @Nizamuddin4493,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlight!
- Good PR ✔️
- Good commits ✔️

# Required Changes ♻️ ⛔️

_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Cathella — CHANGES_REQUESTED** [review]
## 🟡 Status: Changes Requested 🛑

### Hi @Nizamuddin4493 ✋🏻, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️

- [ ] [REQUIRED] I have noticed that when a task is selected, the completed action does not update in the local storage as mentioned by the previous reviewer. This issue is still persistent. I kindly suggest that you look into it and make changes as required 😄.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by oxenprogrammer — APPROVED** [review]
Hi @Nizamuddin4493, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Nizamuddin4493** [comment]
Hi, Reviewer,
I did all my effort on this project. If there is something that is required please mention it in optional. I will do that. because the week is ending in a couple of minutes, if this project will not approved, I will repeat the module

**Pull request merged**